keycloak-event-emitter/src/main/java/io/cloudtrust/keycloak/kafkaeventemitter/SerialisationUtils.java

```diff
@@ -0,0 +1,251 @@
+        if (event.getSessionId() != null) {
+            sessionId = builder.createString(event.getSessionId());
+        }
+
+        // IpAddress
+        int ipAddress = 0;
+        if (event.getIpAddress() != null) {
+            ipAddress = builder.createString(event.getIpAddress());
+        }
+
+        // Error
+        int error = 0;
+        if (event.getError() != null) {
+            error = builder.createString(event.getError());
+        }
+
+        // Details
+        int detailsVec = 0;
+        if (event.getDetails() != null) {
+            List<Integer> tuples = new ArrayList<>();
+            for (Map.Entry<String, String> entry : event.getDetails().entrySet()) {
+                int key = builder.createString(entry.getKey());
+                int value = builder.createString(entry.getValue());
+                int tuple = flatbuffers.events.Tuple.createTuple(builder, key, value);
+                tuples.add(tuple);
+            }
+
+            int[] details = new int[tuples.size()];
+            for (int i = 0; i < tuples.size(); i++) {
+                details[i] = tuples.get(i);
+            }
+
+            detailsVec = flatbuffers.events.Event.createDetailsVector(builder, details);
+        }
+
+        flatbuffers.events.Event.startEvent(builder);
+
+        flatbuffers.events.Event.addUid(builder, uid);
+        flatbuffers.events.Event.addTime(builder, time);
+        flatbuffers.events.Event.addType(builder, type);
+        flatbuffers.events.Event.addRealmId(builder, realmId);
+        flatbuffers.events.Event.addClientId(builder, clientId);
+        flatbuffers.events.Event.addUserId(builder, userId);
+        flatbuffers.events.Event.addSessionId(builder, sessionId);
+        flatbuffers.events.Event.addIpAddress(builder, ipAddress);
+        flatbuffers.events.Event.addError(builder, error);
+        flatbuffers.events.Event.addDetails(builder, detailsVec);
+
+        int flatEvent = flatbuffers.events.Event.endEvent(builder);
+
+        builder.finish(flatEvent);
+
+        return builder.dataBuffer();
+    }
+
+    static ByteBuffer toFlat(ExtendedAdminEvent adminEvent) {
+        FlatBufferBuilder builder = new FlatBufferBuilder(FLATBUFFER_INIT_SIZE);
+
+        // uid
+        long uid = adminEvent.getUid();
+
+        // Time
+        long timeOffset = adminEvent.getTime();
+
+        // RealmId
+        int realmIdOffset = 0;
+        if (adminEvent.getRealmId() != null) {
+            realmIdOffset = builder.createString(adminEvent.getRealmId());
+        }
+
+        // AuthDetails
+        int authDetailsOffset = 0;
+        if (adminEvent.getAuthDetails() != null) {
+            int authDetailsRealmIdOffset = 0;
+            int authDetailsClientIdOffset = 0;
+            int authDetailsUserIdOffset = 0;
+            int authDetailsUsernameAddressOffset = 0;
+            int authDetailsIpAddressOffset = 0;
+
+            if (adminEvent.getAuthDetails().getRealmId() != null) {
```

Comment:
Can't you factorize? (same as previously commented)

Reply:
I did not change anything to serialization